position_in_partition.hh

```diff
@@ -599,22 +599,25 @@ public:
 };
 
 struct view_and_holder {
-    position_in_partition_view view;
-    std::optional<position_in_partition> holder;
+    std::variant<position_in_partition_view, position_in_partition> pos;
+
+    position_in_partition_view view() const {
+        return std::visit(make_visitor([] (position_in_partition_view v) {
+            return v;
+        }), pos);
```

Comment:
Don't we want to create a view if the variant holds a p_i_p?

Reply:
This happens implicitely here, we have implicit conversion from p_i_p to p_i_p_v.